src/nft/utils/txRoute/combineItemsWithTxRoute.ts

```diff
@@ -74,27 +77,33 @@ const itemInRouteAndSamePool = (
 export const combineBuyItemsWithTxRoute = (
   items: UpdatedGenieAsset[],
   txRoute?: RoutingItem[]
-): UpdatedGenieAsset[] => {
-  return items.map((item) => {
+): { hasPriceAdjustment: boolean; updatedAssets: UpdatedGenieAsset[] } => {
+  let hasPriceAdjustment = false
+  const updatedAssets = items.map((item) => {
     const route = getRouteForItem(item, txRoute)
 
-    // if the item is not found in txRoute, it means it's no longer for sale
     if (txRoute && !route) {
       return {
         ...item,
         isUnavailable: true,
       }
     }
 
-    const newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
+    let newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
 
-    // if the price changed
     if (route && 'priceInfo' in route.assetOut) {
-      if (isPriceDiff(newPriceInfo.basePrice, route.assetOut.priceInfo.basePrice)) {
+      const { hasPriceDiff, hasVisiblePriceDiff } = getPriceDiff(
+        newPriceInfo.basePrice,
+        route.assetOut.priceInfo.basePrice
+      )
+
+      newPriceInfo = route.assetOut.priceInfo
```

Comment:
nit: don't think it makes a difference but shouldn't this only be reset if `hasPriceDiff`

Reply:
yes

Reply:
see below basically I think `newPriceInfo` should always be set to `route.assetOut.priceInfo` if it exists as you said. We probably don't need to set `newPriceInfo` initially however just in case some data falls through or something it can't really hurt in my eyes